src/object/schema_class_truncator.cpp

```diff
@@ -0,0 +1,634 @@
+	     * too costly, so we just remove the instances of the current class.
+	     */
+	    error = locator_remove_class_from_index (ws_oid (m_mop), &c->index_btid, &m_class->header.ch_heap);
+	    if (error != NO_ERROR)
+	      {
+		return error;
+	      }
+	  }
+	else
+	  {
+	    /* All the OIDs in the index should belong to the current class, so it is safe to drop and create the
+	     * constraint again. We save the information required to recreate the constraint. */
+
+	    if (SM_IS_CONSTRAINT_UNIQUE_FAMILY (c->type))
+	      {
+		if (sm_save_constraint_info (&m_unique_info, c) != NO_ERROR)
+		  {
+		    return error;
+		  }
+	      }
+	    else if (c->type == SM_CONSTRAINT_FOREIGN_KEY)
+	      {
+		if (sm_save_constraint_info (&m_fk_info, c) != NO_ERROR)
+		  {
+		    return error;
+		  }
+	      }
+	    else
+	      {
+		if (sm_save_constraint_info (&m_index_info, c) != NO_ERROR)
+		  {
+		    return error;
+		  }
+	      }
+	  }
+      }
+
+    return error;
+  }
+
+  /*
+   * drop_saved_constraints () - drop constraints saved in save_constraints_or_clear().
+   *
+   *   return: error code or NO_ERROR
+   *   pred(in): only constraints which meet this condition are processed
+   */
+  int
+  class_truncate_context::drop_saved_constraints (saved_cons_predicate pred)
+  {
+    DB_CTMPL *ctmpl = NULL;
+    SM_CONSTRAINT_INFO *saved = NULL;
+    int error = NO_ERROR;
+
+    /* Even for a class, FK must be dropped earlier than PK, because of the self-referencing case */
+    if (m_fk_info != NULL)
+      {
+	ctmpl = dbt_edit_class (m_mop);
+	if (ctmpl == NULL)
+	  {
+	    assert (er_errid () != NO_ERROR);
+	    error = er_errid ();
+	    return error;
+	  }
+
+	for (saved = m_fk_info; saved != NULL; saved = saved->next)
+	  {
+	    if (!pred (*saved))
+	      {
+		continue;
+	      }
+
+	    error =
+		    dbt_drop_constraint (ctmpl, saved->constraint_type, saved->name, (const char **) saved->att_names, 0);
+	    if (error != NO_ERROR)
+	      {
+		dbt_abort_class (ctmpl);
+		return error;
+	      }
+	  }
+
+	if (dbt_finish_class (ctmpl) == NULL)
+	  {
+	    dbt_abort_class (ctmpl);
+	    assert (er_errid () != NO_ERROR);
+	    error = er_errid ();
+	    return error;
+	  }
+      }
+
+    for (saved = m_unique_info; saved != NULL; saved = saved->next)
+      {
+	if (!pred (*saved))
+	  {
+	    continue;
+	  }
+
+	error = sm_drop_constraint (m_mop, saved->constraint_type,
+				    saved->name, (const char **) saved->att_names, 0, false);
+	if (error != NO_ERROR)
+	  {
+	    return error;
+	  }
+      }
+
+    for (saved = m_index_info; saved != NULL; saved = saved->next)
+      {
+	if (!pred (*saved))
+	  {
+	    continue;
+	  }
+
+	error = sm_drop_index (m_mop, saved->name);
+	if (error != NO_ERROR)
+	  {
+	    return error;
+	  }
+      }
+
+    return error;
+  }
+
+  /*
+   * truncate_heap () - truncate the heap of the class.
+   *
+   *   return: error code or NO_ERROR
+   */
+  int
+  class_truncate_context::truncate_heap ()
```

Comment:
This function looks like it always executes destroying the heap because of the name of this function. Do you have any plan to reconsider when you run the sub-task for DONT_REUSE_OID?

Reply:
I don't think that "truncate" always means "destroying the heap of a class". Doesn't it?

Reply:
@wrlawodms Yes, I mean exactly what you say. But, don't you think the name of this function 'truncate_heap' makes us think "destroying the heap of a class"? How about "truncate_class ()"?

Reply:
Regarding the callees in this function `sm_truncate_using_destroy_heap()` and `sm_truncate_using_delete()`, it doesn't seem to include the context of destroying heap.